src/util.cc

```diff
@@ -222,8 +222,8 @@ void DiagnosticFilename::LocalTime(TIME_TYPE* tm_struct) {
 #ifdef _WIN32
   GetLocalTime(tm_struct);
 #else  // UNIX, OSX
-  struct timeval time_val;
-  gettimeofday(&time_val, nullptr);
+  uv_timeval64_t time_val;
+  CHECK_EQ(0, uv_gettimeofday(&time_val));
   localtime_r(&time_val.tv_sec, tm_struct);
```

Comment:
This changes the type here of `&time_val.tv_sec` from `time_t*` to `int64_t*`, which might not be the same pointer types. Isn’t that an issue? (I’ll kick off CI to see if it’s a problem on one of our supported platforms, but obviously there [are and have been platforms where `time_t` is a 32-bit type](https://en.wikipedia.org/wiki/Year_2038_problem).)

Reply:
Yes, looks like at least the ARM build is failing. It’s probably easiest to keep this as-is?

Reply:
> Yes, looks like at least the ARM build is failing. It’s probably easiest to keep this as-is?

Also macOS

```
20:05:56 ../src/util.cc:227:3: error: no matching function for call to 'localtime_r'
20:05:56   localtime_r(&time_val.tv_sec, tm_struct);
20:05:56   ^~~~~~~~~~~
20:05:56 /Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk/usr/include/time.h:127:12: note: candidate function not viable: no known conversion from 'int64_t *' (aka 'long long *') to 'const time_t *' (aka 'const long *') for 1st argument
20:05:56 struct tm *localtime_r(const time_t * __restrict, struct tm * __restrict);
20:05:56            ^
20:05:56 1 error generated.
```

Reply:
Thank you all for your time to review this :)